drivers/input/input_cst816s.c

```diff
@@ -142,6 +150,13 @@ static int cst816s_process(const struct device *dev)
 		input_report_key(dev, INPUT_BTN_TOUCH, 0, true, K_FOREVER);
 	}
 
+	/* Also put the custom touch gestures into the input queue, 
+	 * some applications may want to process it 
+	 */
+	if (gesture != CST816S_GESTURE_CODE_NONE) {
+		input_report(dev, INPUT_EV_MSC, (uint16_t)gesture, 0, true, K_FOREVER);
```

Comment:
`MSC` is used by linux (just for scancodes), I think we may be better off having a dedicated code for device specific things, how about adding a `INPUT_EV_DEVICE 0xef` just before `INPUT_EV_VENDOR_START` in `include/zephyr/dt-bindings/input/input-event-codes.h`?

Also it may be a good idea to put this behind a Kconfig option, I'd imagine most applications won't use these, save few calls.

Reply:
Thanks @fabiobaltieri .

Just clarifying, when you mention to put the option behind a KConfig option, are you referring to specific driver support or general support of this new code?

Reply:
specific driver